woocommerce/woocommerce-ios
# [Woo POS] M2: Processing Payment UI update (no animation)

Closes: #13343


## Description


Implemented UI updates for Payment Processing views. They come from 2 separate events:
1. displayReaderMessage (with a message sent from the reader ("Remove Card"))
2. processing

The main business logic in this PR is hiding `totalsFields` when a particular payment state is emitted.

I also started implementing some animations but left them outside of this PR since it requires achieving quite a few transitions. It's more complicated than the main logic of this task. It also connects to Cart appearance animations that we haven't yet done so it may be a good time. I created a task: https://github.com/woocommerce/woocommerce-ios/issues/13444. 

A decision to change backgroundColor within `TotalsView` rather than a specific `MessageView` also comes from the learnings when analyzing the transitions. 

## Testing information


1. Launch Woo -> Menu -> Point of Sale Mode
2. Connect Reader
3. Add to Cart
4. Check Out
5. Confirm that Payment Processing - Please wait appears full screen according to designs (TfaZ4LUkEwEGrxfnEFzvJj-fi-2695_105468)
6. Confirm that Payment Processing - Remove card appears full screen according to designs (TfaZ4LUkEwEGrxfnEFzvJj-fi-2530_19752)
7. Confirm that floating button color changed

## Screenshots


https://github.com/user-attachments/assets/696bb0b7-6dc2-41d9-a90e-598015697bf0

---
